### ISSUE TITLE
Add size field in the `array`

### ISSUE THREAD

**Issue by kshitij12345** [kshitij12345/CPU-Capability-Dispatcher#5]
Currently our array implementation assumes 1-d array. Adding a size field will allow us to store multi-dim arrays.

(We might also have to add strides optionally)

cc: @krshrimali

**Comment by kshitij12345** [comment]
@krshrimali would you like to give it a shot?

**Comment by krshrimali** [comment]
Yes, allow me one day to scope this - and I'll get back to you here. @kshitij12345

**Comment by kshitij12345** [comment]
Yes take your time. There is no hard deadline.
